packages/driver/src/cy/commands/actions/select.js

```diff
@@ -150,6 +150,16 @@ module.exports = (Commands, Cypress, cy) => {
           }
         })
 
+        _.each(optionEls, ($el) => {
+          if ($el.parent('optgroup').prop('disabled')) {
```

Comment:
```suggestion
          if ($el.closest('optgroup').prop('disabled')) {
```

It might not be valid html, but I tried it out and it's possible to have an `<option>` nested deeper inside an `optgroup`, so we should guard against that and not assume it's a direct child.